src/common/snippets/docs/mha_optimization_guide.md

```diff
@@ -0,0 +1,81 @@
+2. Tokenization restrictions for performance reasons: executing MHA as separated operations may be faster in some cases.
+
+The CPU plugin callback for TokenizeMHASnippets is placed in [transformation_pipeline.cpp](../../../plugins/intel_cpu/src/transformations/transformation_pipeline.cpp) file (please see the code in `MainSnippets` method).
+
+**Please note that the CPU callback is usually ignored in cpu functional tests.**
+
+## Snippets Common Optimizations
+
+After tokenization, snippets [common optimizations](../src/pass/common_optimizations.cpp) are applied to the tokenized Subgraphs. These transformations can modify both the Subgraph's body and its surroundings (e.g. extract constant nodes outside the Subgraph). Let's explore several transformations that can impact MHA performance.
+
+### ExtractUnsupportedTransposes
+
+[ExtractUnsupportedTransposes](../src/pass/extract_unsupported_transposes.cpp) moves up unsupported Transposes outside the Subgraph. **Please note: the "unsupported" Transpose actually can be executed via Snippets, however CPU plugin implementation usually works faster.**
+
+### SplitDimensionM
+
+[SplitDimensionM](../src/pass/split_dimension_m.cpp) splits M dimension of MHA on 2 parts (`batch_m` and `new_m`) by inserting Reshape on A input of the first Matmul and output of the second Matmul (the rest Subgraph's inputs are reshaped by Unsqueeze-like reshapes in order not to break subgraph semantic). This allows to increase parallel work amount due to `batch_m`. The splitting is performed based on heuristic algorithm which can be found in `SplitDimensionM::get_splited_dimensions` method.
+
+**Important notes:**
+- Since `SplitDimensionM` depends on parallel concurrency (number of threads which are used in one stream), the transformation result depends not only on the HW platform, but on number of streams used during model inference as well. For instance, this might lead to different result in throughput and latency hint modes.
+- `SplitDimensionM::can_be_optimized` is used in CPU plugin callback: if this method reports that appropriate parallel work amount can not be set for the MHA, the tokenization doesn't happen.
+
+## Brgemm Blocking
+
+Within the Snippets CPU backend, the MatMul is executed using the Brgemm primitive. For enhancing the execution efficiency, blocking across the M, K, and N matmul dimensions is used.
+
+### Blocking Parameters
+
+The heuristics for determining the optimal block sizes can be found in [SetBrgemmCPUBlockingParams](../../../plugins/intel_cpu/src/transformations/snippets/x64/pass/set_brgemm_cpu_blocking_params.cpp).
+
+**Important notes:**
+- Blocking by M dimension is shared between both Brgemms. Please see [SplitLoops](../include/snippets/lowered/pass/split_loops.hpp) lowered pass for the details.
+- The heuristics from `SetBrgemmCPUBlockingParams` are not currently applied by K and N dimensions for BF16/INT8 Brgemms, although it is technically supported.
```

Comment:
I don't think that we need to point out all todos and technical limitations of current pipeline. This would not make the docs any more readable.
Let's remove this bullet (The heuristics...) and merge the previous one with the paragraph before Important Notes

Reply:
Done, please check